### PR TITLE
Forward rel/6.4 to rel/7.0

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Forward port from rel/6.4 to rel/7.0
Removes rel/6.4 getLocationTemp.js

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/bucket/getLocationTemp.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import assert from 'assert';
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/bucket/getLocationTemp.js]: the commit that introduced this test should not be forward ported.  this test was just for 6.4.

**Comment by alexanderchan-scality** [comment]
@ironman-machine try

File: tests/unit/metadata/bucketclient/backend.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+const assert = require('assert');
+const http = require('http');
+const BucketClientInterface =
+    require('../../../../lib/metadata/bucketclient/backend').default;
```
Comment (ThibaultRiviere) [inline tests/unit/metadata/bucketclient/backend.js]: default?

Comment (alexanderchan-scality) [reply tests/unit/metadata/bucketclient/backend.js]: Nice catch. Didn't see that when trying to forward 6.4 to 7.0

**Comment by alexanderchan-scality** [comment]
Effectively, yes. The result is from merging 6.4 to 7.0, and needing to drop some commits as they are no longer required.

**Comment by ThibaultRiviere** [comment]
Without using merge you will need to to it each time you want to forward something from `6.4` or using only cherry-pick  now.

**Comment by LaurenSpiegel** [comment]
@ironman-machine try

**Pull request merged**